Modules/_datetimemodule.c

```diff
@@ -3225,6 +3227,174 @@ date_isoweekday(PyDateTime_Date *self, PyObject *Py_UNUSED(ignored))
     return PyLong_FromLong(dow + 1);
 }
 
+PyDoc_STRVAR(iso_calendar_date__doc__,
+"The result of date.isocalendar() or datetime.isocalendar()\n\n\
+This object may be accessed either as a tuple of\n\
+  ((year, week, weekday)\n\
+or via the object attributes as named in the above tuple.");
+
+typedef struct {
```

Comment:
I have not spent enough time trying to implement this to know, but is there a reason why you've implemented a whole new class here rather than creating a `StructSeq` and then subclassing that rather than subclassing `tuple` directly?

Admittedly it's a bit neater to not complicate the MRO, but it does end up with a lot more implementation code when we really just want to override `__reduce__`.

Reply:
@pganssle 
I actually intend to subclassing tuple directly.
However, as I am not an expert of C API extension. 
if this is the wrong way, please let me know the standard way of subclassing in C API

I am just overriding tuple with
__repr__ -> iso_calendar_date_repr
__reduce__ -> isocalendardate_reduce
__getset__ -> iso_calendar_date_methods
as same as pure python version.

Reply:
You can see IsoCalendarDate support basic tuple's operation without any define

Reply:
Yes I noticed you are subclassing `tuple` directly, but that means you have to re-implement `StructSequence` yourself here. It's not a terribly complicated thing to implement, so it could be a reasonable thing to do for many reasons, I just wanted to check if you had deliberately ruled out using a StructSequence as the base instead of `tuple`.

Reply:
reference https://docs.python.org/3/extending/newtypes_tutorial.html#subclassing-other-types

Reply:
@pganssle 
Oh, now I understood what you want to say (Sorry for my poor English)
As you know when we create this type with structsequence there was a test failure with test_os.
So I try to re-implement it as same as pure python and I notice that it works well.

> that means you have to re-implement StructSequence yourself here.

Yes, this some kinds of complicated works but since we re-implement named tuple in pure python version. this works looks like same work around.

Reply:
Can't reuse structseq type but override the type __reduce__ method?

Reply:
@vstinner I am going to find a way to do this.

Reply:
cc @pablogsal

Reply:
If you write a PR using structseq, I suggest to create a separated PR. So we can compare both approaches. Create [WIP] PR without overriding `__reduce__()` if you want help.

Reply:
Great idea

Reply:
@vstinner @pablogsal 
https://github.com/python/cpython/pull/17651 is the experimental PR for comparing both approach.
I'd like to discuss how we will finalize this PR. (Do not change is the one of the options.)